tmt/steps/prepare/artifact/__init__.py

```diff
@@ -53,15 +54,32 @@ def go(
     ) -> PluginOutcome:
         outcome = super().go(guest=guest, environment=environment, logger=logger)
         # TODO: Get and handle repositories
-        # TODO: Create the local repository
         for raw_provider_id in self.data.provide:
             provider_class = get_artifact_provider(raw_provider_id)
             provider_id_sanitized = tmt.utils.sanitize_name(raw_provider_id, allow_slash=False)
             logger = self._logger.descend(raw_provider_id)
             provider = provider_class(raw_provider_id, logger=logger)
             download_path = self.plan_workdir / "artifacts" / provider_id_sanitized
             # TODO: Not using exclude_pattern yet.
-            provider.fetch_contents(guest, download_path)
+            downloaded_paths = provider.fetch_contents(guest, download_path)
```

Comment:
This does not seem right especially with the docstring state so far. Let's revisit the design document that we've put together earlier on. That was proposing a completely different method to be defined and used.

Reply:
Implemented the code with the orignal design: 62975af6795462b859785f6a1cce39ff039951db

Reply:
The interface is still not as we would like it to be, particularly copr implementation would break because in that case the provisioner has to create the repo, not the top-level and `fetch_contents` is still ambiguous. I would suggest to instead drop all the return value of `fetch_contents` and add the `contribute_to_the_shared_repo` approach instead. If there are issues with implementing that, let's discuss it here